fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJobBase.java

```diff
@@ -276,12 +277,14 @@ public void updatePartitionTabletMeta(Database db,
                         "Partition[" + partitionName + "] does not exist in table[" + olapTable.getName() + "]");
             }
 
+            WarehouseManager warehouseManager = GlobalStateMgr.getCurrentState().getWarehouseMgr();
             for (PhysicalPartition physicalPartition : partition.getSubPartitions()) {
                 for (MaterializedIndex index : physicalPartition.getMaterializedIndices(
                         MaterializedIndex.IndexExtState.VISIBLE)) {
                     addDirtyPartitionIndex(physicalPartition.getId(), index.getId(), index);
                     for (Tablet tablet : index.getTablets()) {
-                        Long backendId = Utils.chooseNodeId((LakeTablet) tablet);
+                        Long backendId = warehouseManager.getComputeNodeAssignedToTablet(WarehouseManager.DEFAULT_WAREHOUSE_ID,
```

Comment:
‌‌‌‌Is this BackendId the ID of BE ?  Or is it simply an ID for the node storing a shard in shared-data mode?

Reply:
backendid  for the be storing a shard in shared-data mode